### PR TITLE
Create tests for Rendering, Shaders, Buffers, Exporter, MainWindow, MeasurementDialog

### LINKED ISSUE #302: Create tests that cover exporter.py, rendering.py, shaders.py, buffers,py
Create `test_exporter.py`, `test_rendering.py`, `test_shaders.py`, `test_buffers.py` and design the tests in an analogous way to the existing tests in the `tests/` directory.

### LINKED ISSUE #320: Measurement: errors in dihedral angle calculation
The calculations weren't right. This is going to be fixed by PR #312.

### PR DESCRIPTION
**Pull request by adrianusler**
Closes #302. Ready to merge @Michel-Heinz!
(Also closes #320). 

Tasks still to be finished:
- [x] rendering.py
  - [x] remove_sphere
- [x] buffers.py
- [x] exporter.py
  - [x] general exporter 
- [x] main_window.py
  - [x] show trajectory dialog 
- [x] measuring_tool_dialog.py
  - [x] display angles
  - [x] display dihedral angle

### CODE REVIEW

File: tests/molara/Rendering/test_rendering.py [hunk 1]
```diff
@@ -2,16 +2,19 @@
...
 import numpy as np
+import pytest
 from molara.Rendering.rendering import Renderer
 from PySide6.QtOpenGLWidgets import QOpenGLWidget
 
 if TYPE_CHECKING:
     from pytestqt.qtbot import QtBot
 
 
+@pytest.mark.skipif(sys.platform == "win32")
```
Comment (adrianusler) [inline tests/molara/Rendering/test_rendering.py]: Really nice! Thanks @ab5424!

**Comment by ab5424** [comment]
@adrianusler This looks really nice, a coverage increase of almost 40%-pt.!

**Comment by Michel-Heinz** [comment]
Thanks a lot for worked you put in!

**Comment by ab5424** [comment]
@adrianusler @Michel-Heinz The remaining pre-commit failures are unrelated to this PR (see #321).

**Pull request merged**